src/ArborX_CrsGraphWrapper.hpp

```diff
@@ -0,0 +1,34 @@
+/****************************************************************************
+ * Copyright (c) 2012-2020 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_CRS_GRAPH_WRAPPER_HPP
+#define ARBORX_CRS_GRAPH_WRAPPER_HPP
+
+#include "ArborX_DetailsCrsGraphWrapperImpl.hpp"
+
+namespace ArborX
+{
+
+template <typename ExecutionSpace, typename Tree, typename Predicates,
+          typename CallbackOrView, typename View, typename... Args>
+inline void query_crs(ExecutionSpace const &space, Tree const &tree,
+                      Predicates const &predicates,
+                      CallbackOrView &&callback_or_view, View &&view,
+                      Args &&... args)
+{
+  Details::CrsGraphWrapperImpl::query(
+      space, tree, predicates, std::forward<CallbackOrView>(callback_or_view),
+      std::forward<View>(view), std::forward<Args>(args)...);
```

Comment:
Not sure I understand.

Reply:
AFAICT, this is the only `query_crs` overload/specialization and the only place that calls `Details::CrsGraphWrapperImpl::query`. Hence, I was wondering if the implementation of `Details::CrsGraphWrapperImpl::query` could be moved here.

Reply:
Oh, I see what you are saying. Indeed, it could. I just thought that as this file is user-visible, I would just put the API function here, while hiding all the details in the details file, per usual.